server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -477,88 +477,77 @@ public static boolean isTimeSeriesModeEnabled() {
     }
 
     /**
-     * The {@link IndexMode "mode"} of the index.
+     * in time series mode, the start time of the index, timestamp must larger than start_time
      */
-    public static final Setting<IndexMode> MODE = Setting.enumSetting(
-        IndexMode.class,
-        "index.mode",
-        IndexMode.STANDARD,
-        new Setting.Validator<IndexMode>() {
-            @Override
-            public void validate(IndexMode value) {}
-
-            @Override
-            public void validate(IndexMode value, Map<Setting<?>, Object> settings) {
-                value.validateWithOtherSettings(settings);
-            }
-
-            @Override
-            public Iterator<Setting<?>> settings() {
-                return IndexMode.VALIDATE_WITH_SETTINGS.iterator();
-            }
-        },
```

Comment:
Because I moved the validation around `MODE` has to come after the START_TIME and END_TIME or else it can't validate against them. You get a sneaky NullPointerException on init which was fun to debug.